vm/default_runtime/src/lib.rs

```diff
@@ -410,42 +412,38 @@ pub fn internal_send<ST: StateTree, DB: BlockStore>(
-                    actor::system::Actor.invoke_method(&mut *runtime, *method_num, msg.params())
+                    actor::system::Actor.invoke_method(runtime, *method_num, msg.params())
                 }
                 x if x == *INIT_ACTOR_CODE_ID => {
-                    actor::init::Actor.invoke_method(&mut *runtime, *method_num, msg.params())
+                    actor::init::Actor.invoke_method(runtime, *method_num, msg.params())
                 }
                 x if x == *CRON_ACTOR_CODE_ID => {
-                    actor::cron::Actor.invoke_method(&mut *runtime, *method_num, msg.params())
+                    actor::cron::Actor.invoke_method(runtime, *method_num, msg.params())
                 }
                 x if x == *ACCOUNT_ACTOR_CODE_ID => {
-                    actor::account::Actor.invoke_method(&mut *runtime, *method_num, msg.params())
+                    actor::account::Actor.invoke_method(runtime, *method_num, msg.params())
                 }
                 x if x == *POWER_ACTOR_CODE_ID => {
-                    actor::power::Actor.invoke_method(&mut *runtime, *method_num, msg.params())
+                    actor::power::Actor.invoke_method(runtime, *method_num, msg.params())
                 }
                 x if x == *MINER_ACTOR_CODE_ID => {
-                    // not implemented yet
-                    // actor::miner::Actor.invoke_method(&mut *runtime, *method_num, msg.params())
-                    todo!()
+                    actor::miner::Actor.invoke_method(runtime, *method_num, msg.params())
                 }
                 x if x == *MARKET_ACTOR_CODE_ID => {
-                    // not implemented yet
-                    // actor::market::Actor.invoke_method(&mut *runtime, *method_num, msg.params())
-                    todo!()
+                    actor::market::Actor.invoke_method(runtime, *method_num, msg.params())
                 }
                 x if x == *PAYCH_ACTOR_CODE_ID => {
-                    actor::paych::Actor.invoke_method(&mut *runtime, *method_num, msg.params())
+                    actor::paych::Actor.invoke_method(runtime, *method_num, msg.params())
                 }
                 x if x == *MULTISIG_ACTOR_CODE_ID => {
-                    actor::cron::Actor.invoke_method(&mut *runtime, *method_num, msg.params())
+                    actor::cron::Actor.invoke_method(runtime, *method_num, msg.params())
                 }
                 x if x == *REWARD_ACTOR_CODE_ID => {
-                    actor::cron::Actor.invoke_method(&mut *runtime, *method_num, msg.params())
+                    actor::cron::Actor.invoke_method(runtime, *method_num, msg.params())
                 }
                 _ => Err(ActorError::new(
-                    ExitCode::SysErrForbidden,
-                    "invalid method id".to_owned(),
+                    ExitCode::SysErrorIllegalActor,
+                    format!("no code for actor at address {}", msg.to()),
```

Comment:
Since you are updating the exit code here, I think the `transfer` map_err exit code on ~LN 401, near the start of this fn should be changed from `ExitCode::SysErrInternal` to `ExitCode::SysErrActorNotFound`.

Reply:
Also updated the names of the exit codes to match 1:1, not sure if changed since eric updated